### ISSUE TITLE
CPU 100% when use Q.nbind()

### ISSUE THREAD

**Issue by yetithefoot** [kriskowal/q#725]
I'm using https://github.com/google/google-api-nodejs-client for my app.
When im trying to wrap and run ```drive.files.list``` with ```Q.nbind()``` it never ends and stress CPU to 100% somewhere in array.reduce.

Callback from ```drive.files.list``` returns 3 params (err, data, request), but first two of them are: err, data as generic node-styled callback.

Q version 1.4.1.

**Comment by domenic** [comment]
Can you try doing this without Q and seeing if the CPU usage remains high?

**Comment by yetithefoot** [comment]
Yep! Everything is OK without Q. I can send you arguments from my response.

**Comment by yetithefoot** [comment]
@domenic Also everything is OK if i use ```var dfd = Q.defer()``` and then ```dfd.resolve()/reject()```.
So the problem is somwhere inside Q library for this concrete case.

**Comment by domenic** [comment]
Can you create a reproduction step without google-api-nodejs-client?

**Comment by kriskowal** [comment]
I have a big request, if you can pull it off. A snapshot of a sample of stacks would be very useful to us, and a very powerful tool for you to use to debug similar issues going forward. What operating system are you using? If you are running on Linux, you could try SystemTap with [node-stap][] to capture a flame graph. There are similar tools for other operating systems.

[node-stap]: https://github.com/uber/node-stap
